### ISSUE TITLE
pre-open - intercept a request to open an image?

### ISSUE THREAD

**Issue by robhicks** [openseadragon/openseadragon#969]
Is there a way to intercept and prevent a request to open an image? 

Imagine a user viewing an image with a navigator. The user is presented with an overlay to input data about the image. When the user attempts to go to a different image, the user may need to be warned about losing data and the default action of opening the image stopped.

**Comment by iangilman** [comment]
Sounds like you're talking about the next/previous buttons for sequenceMode? There is not yet such a mechanism. Looks like a good place to add it would be the `page` event fired by `goToPage`:

https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L1772

We'd want to provide a mechanism for anyone receiving that event to be able to tell it to cancel itself.

Is this something you might be up for working on?

**Comment by robhicks** [comment]
I'm up for it. I assume it would work in a fashion similar to pre-full-page. As far a cancelling it, I imagined the handler would catch the event and then based upon app logic change the preventDefaultAction to false. But admittedly, I'm not entirely sure how pre-full-page works, particularly how the handler passes the fullPageEventArgs. I looked at the tests to see how works but didn't see anything. Can you point me to an example.

In a related question, if a user is in world view, is there an event raised when the user clicks on a particular image? I think it would also be important to be able to know which image was click so that data could be displayed about that image.

**Comment by iangilman** [comment]
Yes, pre-full-page's preventDefaultAction seems like a good model to follow. Here's the code for that: 

https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L881-L899

We actually don't yet have a way to know which image was clicked. The process is to get the click location in viewport coordinates and then run through all of the images in the world and ask for their bounds and check the location against that.

It seems a nice addition to the canvas-click event would be to include this information. If you're up for adding that as well, that would be lovely!

**Comment by robhicks** [comment]
I'll work on it.
